From a9434be0ae335a726ea3dc93ffbc134a5e8acf2c Mon Sep 17 00:00:00 2001
From: Andrej Koehler <Andrej.Koehler@garmin.com>
Date: Mon, 17 Aug 2026 14:22:56 +0200
Subject: [PATCH] test_concurrent_futures: skip flaky tests for 3.8

test_concurrent_futures tests:
 - ProcessPoolForkserverWaitTest.test_first_exception
 - ProcessPoolSpawnWaitTest.test_first_exception
are failing on Python 3.8, probably QEMU related.

This is potentially related to CPython issue GH-91607:
https://github.com/python/cpython/issues/91607
which cleaned up this test suite in Python 3.9.

Python 3.8 cannot receive this test-suite correction.

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Andrej Koehler <akhldev@proton.me>
---
 Lib/test/test_concurrent_futures.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Lib/test/test_concurrent_futures.py b/Lib/test/test_concurrent_futures.py
index a5a746e..829dde5 100644
--- a/Lib/test/test_concurrent_futures.py
+++ b/Lib/test/test_concurrent_futures.py
@@ -476,6 +476,9 @@ class WaitTests:
         self.assertEqual(set([future1]), pending)
 
     def test_first_exception(self):
+        if getattr(self, "ctx", None) in ("forkserver", "spawn"):
+            self.skipTest("flaky under QEMU on Python 3.8")
+
         future1 = self.executor.submit(mul, 2, 21)
         future2 = self.executor.submit(sleep_and_raise, 1.5)
         future3 = self.executor.submit(time.sleep, 3)
